firedrake/solving_utils.py

```diff
@@ -229,10 +266,38 @@ def split(self, fields):
                         W = V.sub(field_renumbering[index])
                     if cmpt is not None:
                         W = W.sub(cmpt)
-                    bcs.append(type(bc)(W,
-                                        bc.function_arg,
-                                        bc.sub_domain,
-                                        method=bc.method))
+                    if isinstance(bc, DirichletBC):
+                        return DirichletBC(W,
+                                           bc.function_arg,
+                                           bc.sub_domain,
+                                           method=bc.method)
+                    elif isinstance(bc, EquationBC):
+                        bc_F = replace(splitter.split(bc.F, argument_indices=(field, )), {bc.u: u})
+                        bc_J = replace(splitter.split(bc.J, argument_indices=(field, field)), {bc.u: u})
+                        bc_Jp = None if bc.Jp is None else replace(splitter.split(bc.Jp, argument_indices=(field, field)), {bc.u: u})
+                        ebc = EquationBC(bc_F == 0,
+                                         subu,
+                                         bc.sub_domain,
+                                         method=bc.method,
+                                         bcs=None,
+                                         J=bc_J,
+                                         Jp=None if bc.Jp_eq_J else bc_Jp,
```

Comment:
Just make this `bc_Jp` and the assignment of `bc_Jp` something like `bc_Jp = None if (bc.Jp_eq_J or bc.Jp is None) else ...` ?

Reply:
Thinking about this, I realised that `bc.Jp` is never `None`.  Simplified now.